chainntnfs/txconfnotifier.go

```diff
@@ -403,12 +634,25 @@ out:
 
 	// Then, we'll dispatch notifications for all the transactions that have
 	// become confirmed at this new block height.
-	for ntfn := range tcn.ntfnsByConfirmHeight[tcn.currentHeight] {
+	for ntfn := range tcn.ntfnsByConfirmHeight[blockHeight] {
+		confSet, ok := tcn.confNotifications[*ntfn.TxID]
+		if !ok {
+			Log.Warnf("Conf set not found for scheduled dispatch "+
+				"of txid=%x", ntfn.TxID)
+			continue
+		}
+
+		if confSet.details == nil {
```

Comment:
I would actually prefer just to use the details directly without checking for `nil` here and other places where it should be set. If that happens it is a quite serious bug and will be buried in the logs if we just print and continue.